ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/util/DnUtils.java

```diff
@@ -0,0 +1,44 @@
+package org.ccci.idm.user.ldaptive.dao.util;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import org.ccci.idm.user.Dn;
+import org.ldaptive.DnParser;
+import org.ldaptive.LdapAttribute;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class DnUtils {
+    private static final String DELIMITER = ",";
+    private static final String VALUE_DELIMITER = "=";
+
+    @Nonnull
+    public static Dn toDn(@Nullable final String rawDn) {
+        if (rawDn == null) {
+            return Dn.ROOT;
+        }
+
+        final ImmutableList.Builder<Dn.Component> builder = ImmutableList.builder();
+        for (final LdapAttribute attribute : Lists.reverse(DnParser.convertDnToAttributes(rawDn))) {
+            builder.add(new Dn.Component(attribute.getName(), attribute.getStringValue()));
+        }
+        return new Dn(builder.build());
+    }
+
+    @Nonnull
+    public static String toString(@Nonnull final Dn dn) {
+        final StringBuilder sb = new StringBuilder();
+
+        // append components
+        for (final Dn.Component component : Lists.reverse(dn.getComponents())) {
+            if (sb.length() > 0) {
```

Comment:
for 1 I probably prefer (b). generally you already know what type of object you have before you get to the transformer, so repeating that in the method name I feel is redundant

2 is a toss up, I think I generally prefer having them on the object itself for encapsulation. and sometimes they end up as static methods on the target object class (e.g. `Obj1.fromObj2(obj2)`)